### PR TITLE
Add TI-DeepONet DPC heat notebook example

### PR DESCRIPTION
**Pull request by Parv621**
# Notebook: Add TI-DeepONet DPC heat notebook example

## Summary
- Add a differentiable predictive control (DPC) notebook for the heat equation using a trained TI-DeepONet model within Neuromancer. Demonstrates end-to-end policy learning via long-horizon rollouts with differentiable physics.

## Changes
- `examples/control/Part_7_TIDON_DPC_HEAT.ipynb` adds a DPC example for PDE control based on TI-DeepONet, including problem setup, policy parameterization, rollout-based loss construction, and training using Neuromancer.

## Context
- Recreates the DPC framework for PDE control using TI-DeepONet (time-integrated Deep Operator Network), following the methodology in prior DPC literature and the reference implementation in PDEControl_DPC.
- Uses a pretrained TI-DeepONet model from `examples/neural_operators/Part_8_TIDON_HEAT.ipynb` as a differentiable surrogate of the heat-equation dynamics.
- The neural operator enables backpropagation through long rollouts, allowing gradients of the control objective with respect to policy parameters.
- The objective is to learn a feedback control policy for a distributed parameter system using differentiable predictive control implemented in Neuromancer.

## References
- Sarkar, D. R., Drgoňa, J., & Goswami, S. (2025). *Learning to Control PDEs with Differentiable Predictive Control and Time-Integrated Neural Operators*. arXiv:2511.08992.

## Testing
- Not run (not needed).

### CODE REVIEW

**Comment by drgona** [comment]
@Parv621 both the file and notebook have import that is not included in the toml file

`import gpytorch`

Do we need this?

**Comment by Parv621** [comment]
> @Parv621 both the file and notebook have import that is not included in the toml file
> 
> `import gpytorch`
> 
> Do we need this?

Added dependency as discussed

**Comment by drgona** [comment]
@Parv621 the examples also contain custom file paths:
`    data_path = "/home/pk222/projects/PDEControl_DPC/datasets/heat_smooth_f_dataset.npz"`

Please remove all unnecessary data loads, the notebooks should be self-contained.
If really necessary to load any sort of file, do it in a similar fashin we handle it in this notebook: 
[neural DAEs](https://colab.research.google.com/github/pnnl/neuromancer/blob/master/examples/DAEs/tank_dae_example.ipynb)

code example:

```
# Raw URL of area.dat
url = "https://raw.githubusercontent.com/pnnl/NeuralDAEs/master/training/area.dat"

# Download and save the file locally
response = requests.get(url)
with open("area.dat", "wb") as f:
    f.write(response.content)
area_data = np.loadtxt('area.dat')
```

**Comment by Parv621** [comment]
> @Parv621 the examples also contain custom file paths: ` data_path = "/home/pk222/projects/PDEControl_DPC/datasets/heat_smooth_f_dataset.npz"`
> 
> Please remove all unnecessary data loads, the notebooks should be self-contained. If really necessary to load any sort of file, do it in a similar fashin we handle it in this notebook: [neural DAEs](https://colab.research.google.com/github/pnnl/neuromancer/blob/master/examples/DAEs/tank_dae_example.ipynb)
> 
> code example:
> 
> ```
> # Raw URL of area.dat
> url = "https://raw.githubusercontent.com/pnnl/NeuralDAEs/master/training/area.dat"
> 
> # Download and save the file locally
> response = requests.get(url)
> with open("area.dat", "wb") as f:
>     f.write(response.content)
> area_data = np.loadtxt('area.dat')
> ```

Added open OneDrive links for the best model

**Comment by drgona** [comment]
@Parv621 

I don't like loading files like this for pre-trained model. Lets have system ID done in the script and notebook before training DPC.

These paths are undefined in the notebook:
path_to_model = "path/to/model_best.pt"

**Comment by Parv621** [comment]
@jdrgona Updated the notebooks and scripts for your review
